src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -108,219 +287,398 @@ public virtual async Task InvokeAsync(HttpContext context, RequestDelegate next)
-                OperationName = urlGQLRequest.OperationName ?? bodyGQLRequest?.OperationName
+                Query = urlGQLRequest?.Query ?? bodyGQLRequest?.Query,
+                Variables = urlGQLRequest?.Variables ?? bodyGQLRequest?.Variables,
+                Extensions = urlGQLRequest?.Extensions ?? bodyGQLRequest?.Extensions,
+                OperationName = urlGQLRequest?.OperationName ?? bodyGQLRequest?.OperationName
             };
-        }
 
-        // Prepare context and execute
-        var userContextBuilder = context.RequestServices.GetService<IUserContextBuilder>();
-        var userContext = userContextBuilder == null
-            ? new Dictionary<string, object>() // in order to allow resolvers to exchange their state through this object
-            : await userContextBuilder.BuildUserContext(context);
+            // Prepare context and execute
+            await HandleRequestAsync(context, _next, gqlRequest);
+        }
+        else if (Options.EnableBatchedRequests)
+        {
+            await HandleBatchRequestAsync(context, _next, bodyGQLBatchRequest);
+        }
+        else
+        {
+            await HandleBatchedRequestsNotSupportedAsync(context, _next);
+        }
+    }
 
-        var executer = context.RequestServices.GetRequiredService<IDocumentExecuter<TSchema>>();
-        await HandleRequestAsync(context, next, userContext, bodyGQLBatchRequest, gqlRequest, executer, cancellationToken);
+    /// <summary>
+    /// Perform authentication, if required, and return <see langword="true"/> if the
+    /// request was handled (typically by returning an error message).  If <see langword="false"/>
+    /// is returned, the request is processed normally.
+    /// </summary>
+    protected virtual async ValueTask<bool> HandleAuthorizeAsync(HttpContext context, RequestDelegate next)
+    {
+        var success = await AuthorizationHelper.AuthorizeAsync(
+            new AuthorizationParameters<(GraphQLHttpMiddleware Middleware, HttpContext Context, RequestDelegate Next)>(
+                context,
+                Options,
+                static info => info.Middleware.HandleNotAuthenticatedAsync(info.Context, info.Next),
+                static info => info.Middleware.HandleNotAuthorizedRoleAsync(info.Context, info.Next),
+                static (info, result) => info.Middleware.HandleNotAuthorizedPolicyAsync(info.Context, info.Next, result)),
+            (this, context, next));
+
+        return !success;
     }
 
+    /// <summary>
+    /// Perform authorization, if required, and return <see langword="true"/> if the
+    /// request was handled (typically by returning an error message).  If <see langword="false"/>
+    /// is returned, the request is processed normally.
+    /// <br/><br/>
+    /// By default this does not check authorization rules becuase authentication may take place within
+    /// the WebSocket connection during the ConnectionInit message.  Authorization checks for
```

Comment:
```suggestion
    /// By default this does not check authorization rules because authentication may take place within
    /// the WebSocket connection during the connection_init message.  Authorization checks for
```

Reply:
The correct name of the message is ConnectionInit.

https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#connectioninit

Reply:
```suggestion
    /// By default this does not check authorization rules because authentication may take place within
    /// the WebSocket connection during the ConnectionInit message.  Authorization checks for
```

Reply:
Once the updated websocket code it merged in, it will match the code also.

Reply:
Merged the misspelling.

Reply:
Note that for the old protocol, the correct name of the message is GQL_CONNECTION_INIT.  This also is very clear in the updated websocket code.

https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md